sys/aarch64/rootdev.c

```diff
@@ -45,11 +77,52 @@ static bus_space_t *rootdev_bus_space = &(bus_space_t){
 
 static void rootdev_intr_setup(device_t *dev, unsigned num,
                                intr_handler_t *handler) {
-  /* TODO(pj) */
+  rootdev_t *rd = dev->parent->state;
+
+  assert(num < N_IRQ);
+  intr_event_t *event = &rd->intr_event[num];
+  intr_event_add_handler(event, handler);
 }
 
 static void rootdev_intr_teardown(device_t *dev, intr_handler_t *handler) {
-  /* TODO(pj) */
+  intr_event_remove_handler(handler);
+}
+
+/* Read 32 bit pending register located at va and run handlers. */
+static inline void intr_handle(vaddr_t va, intr_event_t *events) {
```

Comment:
Does compiler honour `inline` in case of function as large as this one?

Reply:
Could you pass a pointer that has a meaningful name instead of just `va` ?

Reply:
Yes, compiler honour `inline` in that case.